Changes from Open Wide fork

franck-grenier opened the pull request:
Hello Brookins Consulting,
here is a big pull request from our master branch.

cfeed02 and 3a5e8ae have already been integrated on your side.

I let you check the next commits.

brookinsconsulting commented:
Hello @franck-grenier 

Thank you for submitting this PR. We are very pleased to be able once again to collaborate with Open-Wide on this extension.

Because your fork was not current and because of a large number of concerns with the implementation of some of your changes an automatic merge was not possible.

We took this morning to work through each of the changes, within each of the files, within each  commits to address the underlying need for change. Many changes were made via direct cherry picking (where possible). Other changes which required refactoring where made via cherry picking + BC modifications.

Nearly every change presented in this PR was either implemented or re-implemented to preserve BC, CS and user choice (this project is extreme 3.x legacy so we do not remove features as much as possible and instead make them ini setting controllable).

Please remember in the future when making changes that bccie uses the eZ Publish Legacy Code Standard and not PSR/PSR2 or eZ Publish 5 Code Standard. Noted that there are lots of older code we avoid refactoring the CS for stability reasons (even when in violation of eZPLCS). We had to refactor quite a bit of your PR commits for CS. Please try to learn from the differences of your PR CS and the re-implemented code CS as a guide of what we are trying to maintain in bccie.

Thank you again so very much for your hard work, collaboration and contributions to bccie!

We have updated the latest 1.1.2 changelog and credits documentation to cover almost all of the related changes. Please review the updated documentation and master commit history to familiarize yourself with the latest changes.

Also if you would, it would be a great help to BC, Open-Wide and the community if you were to strongly consider merging over (replacing) your previous changes implementation in favor of BC's implementation. 

We took great care to ensure your needs were provided for within our implementation and not leave out any changes you might require. 

Using the upstream re-implementation to replace your own will help ensure the bccie community does not suffer from 'fragmentation' between forks on GitHub or elsewhere within the eZ Community and also ensures we can all in the future quickly and cleanly merge changes between our forks and the upstream repository.

That said, this PR has be manually re-implemented and will now be closed.

Cheers,
Brookins Consulting

franck-grenier commented:
Hello Brookins Consulting,
thank you for handling this big PR. I guess it was not easy work...

We know what we fixed and changed in our fork. So, we'll migrate to your last version and we'll test it in order to ensure that it contains each of our changes.

Thank you for maintening this very useful extension !

brookinsconsulting commented:
Hello @franck-grenier 

You are very welcome. We were happy to complete the work required to ensure both our repos of bccie can be in sync and prevent fragmentation with the bccie community.

Though to be honest we spent like .. over 5 hours with the work, it was time consuming but well worth the results.

We will wait for you to make a future comment here saying your synced up and need no further changes at this time and then we will make an official build (downloadable packages hosted on projects.ez.no and a github tag release).

If by chance you need anything or have more questions; please feel free to ask.

BTW, We sent you an email using the contact info from your personal website. If you could please confirm your email by sending us an email reply just in case we need to correspond directly in the future.

Cheers,
Brookins Consulting
